CodeU_Assignment_6/src/codeu_assignment_6/KemRearrangingCars.java

```diff
@@ -0,0 +1,129 @@
+        for (int i=0; i<n; i++) {
+            if (lotCurrState[i] != permutationB[i]){
+                /* If location is not free, move the item there */
+                currZeroLocation = swap(lotCurrState[i], currZeroLocation, currIndexToElements, lotCurrState, moveSeries);
+
+                /* swap the right item into location i (excpet in the case of 0) */
+                currZeroLocation = swap(permutationB[currZeroLocation], currZeroLocation, currIndexToElements, lotCurrState, moveSeries);
+            }
+        }
+        
+        /* The return value is pairs of moves ordered from 0 to size of the array */
+        return moveSeries;
+    }
+    
+    /**
+     * Return false if the input is invalid (null, not the same size...) 
+     * 
+     * If required, you may add here throw of exceptions
+     */
+    private boolean isValidInput(Integer[] permutationA, Integer[] permutationB) {
+        if ((permutationA == null) || (permutationB == null)) {
+            return false;
+        }
+        if (permutationA.length != permutationB.length) {
+            return false;
+        }
+        for (int i=0;i<permutationA.length;i++) {
+            if (permutationA[i] >= permutationA.length) {
+                return false;
+            }
+            if (permutationB[i] >= permutationB.length) {
+                return false;
+            }
+        }
+        
+        return true;
+    }
+    
+    /**
+     * Process: swap two items in location of the currToSwap and the free space
+     * @return the location of the free space after the swap
+     */
+    private int swap(int currToSwap, int currZeroLocation, 
+            Integer[] currIndexToElements, Integer[] lotCurrState, ArrayList<Pair<Integer,Integer>> moveSeries) {
+        if (currToSwap == 0) {
+            return currZeroLocation;
+        }
+        
+        /* Find the location of the item to swap to current location */
+        int locationCurrToSwap = currIndexToElements[currToSwap];
+                    
+        /* Swap! */
+        lotCurrState[currZeroLocation] = currToSwap;
+        lotCurrState[locationCurrToSwap] = 0;
+
+        /* Update the output moves */
+        Pair<Integer,Integer> move = new Pair<>(locationCurrToSwap, currZeroLocation);
+        moveSeries.add(move);
+
+        /* Update the indeces */
+        currIndexToElements[currToSwap] = currZeroLocation;
+        currIndexToElements[0] = locationCurrToSwap;
+
+        /* Will do out: currZeroLocation = locationCurrToSwap; */
+        return locationCurrToSwap;
+    }
+    
+    public ArrayList<Pair<Integer,Integer>> getMovesRearrangingCarsAndPrint(Integer[] permutationA, Integer[] permutationB) {
+        ArrayList<Pair<Integer,Integer>> moves = getMovesRearrangingCars(permutationA, permutationB);
+         
+        if (moves != null) {
+            moves.forEach(t->System.out.println("move from " + t.getKey() + " to " + t.getValue()));
+        }
+         
+        return moves;
+    }
+}

```

Comment:
Nice and simple algorithm. You can complete the task in fewer moves - closer to N than 2*N - so it might be worth having a quick think about that, but generally this is good, clean, simple code. Thank you.